tools/fw-update/rs-fw-update.cpp

```diff
@@ -119,6 +119,40 @@ void list_devices(rs2::context ctx)
     }
 }
 
+int write_to_mipi_device( const std::vector< uint8_t > & fw_image)
+{
+    // Write firmware to appropriate file descritptor
+    std::cout << std::endl << "Update can take up to 2 minutes" << std::endl;
+    std::ofstream fw_path_in_device( "/dev/d4xx-dfu504", std::ios::binary );
```

Comment:
If an exception occur no file close will happen.
Normally to overcome it we use a unique_ptr with a deleter that close the file when our of scope.
Use can find examples in the code